DataFormats/simulation/include/SimulationDataFormat/IOMCTruthContainerView.h

```diff
@@ -0,0 +1,105 @@
+// Copyright CERN and copyright holders of ALICE O2. This software is
+// distributed under the terms of the GNU General Public License v3 (GPL
+// Version 3), copied verbatim in the file "COPYING".
+//
+// See http://alice-o2.web.cern.ch/license for full licensing information.
+//
+// In applying this license CERN does not waive the privileges and immunities
+// granted to it by virtue of its status as an Intergovernmental Organization
+// or submit itself to any jurisdiction.
+
+/// \file IOMCTruthContainerView.h
+/// \brief A special IO container - splitting a given vector to enable ROOT IO
+/// \author Sandro Wenzel - August 2020
+
+#ifndef ALICEO2_DATAFORMATS_IOMCTRUTHVIEW_H_
+#define ALICEO2_DATAFORMATS_IOMCTRUTHVIEW_H_
+
+#include "GPUCommonRtypes.h" // to have the ClassDef macros
+#include <vector>
+#include <gsl/span>
+
+namespace o2
+{
+namespace dataformats
+{
+
+///
+/// A specially constructed class allowing to stream a very large
+/// vector buffer to a ROOT file. This is needed since ROOT currently has a size
+/// limitation of ~1GB for data that it can stream per entry in a branch.
+/// The solution is based on the ability of ROOT to split entries per data member, so
+/// some input buffer gets divided into multiple parts.
+///
+/// TODO: We could template this class to encode original type information (for the input buffer).
+class IOMCTruthContainerView
```

Comment:
The name could be more descriptive (as you already said). I think it should point out clearly that this is only for root serialization of large buffers.

I don't quite like the fact that the 10 buffers are explicit. Couldn't we make this a template parameter somehow? And I think there should be a check that we remain in the ROOT-compatible buffer size, to get a proper error when we overrun the buffer.

Reply:
I agree. Not sure how a template param is possible ... but some preproc macro expansion was on my list anyway.
My plan is to make these improvements in another round, once we have experience with a concrete detector use-case.